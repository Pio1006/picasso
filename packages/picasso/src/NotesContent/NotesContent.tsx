import React, { forwardRef, ReactNode } from 'react'
import { BaseProps } from '@toptal/picasso-shared'

import Typography from '../Typography'

export interface Props extends BaseProps {
  children: ReactNode
}

export const NotesContent = forwardRef<HTMLDivElement, Props>(
  function NotesContent ({ children, ...rest }, ref) {
    return (
      <Typography ref={ref} as='p' size='medium' color='black' {...rest}>
        {children}
      </Typography>
    )
  }
)

NotesContent.defaultProps = {}

NotesContent.displayName = 'NotesContent'

export default NotesContent
