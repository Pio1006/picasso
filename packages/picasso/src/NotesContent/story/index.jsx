import PicassoBook from '~/.storybook/components/PicassoBook'
import { NotesContent } from '../NotesContent'

const componentDocs = PicassoBook.createComponentDocs(
  NotesContent,
  'Notes.Content',
  'Notes content'
)

export default {
  componentDocs
}
