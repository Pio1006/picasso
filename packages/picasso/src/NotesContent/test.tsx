import React, { ReactNode } from 'react'
import { render } from '@toptal/picasso/test-utils'

import NotesContent from './NotesContent'

const renderNotesContent = (children: ReactNode) => {
  return render(<NotesContent>{children}</NotesContent>)
}

describe('NotesContent', () => {
  it('default render', () => {
    const { container } = renderNotesContent('content')

    expect(container).toMatchSnapshot()
  })
})
