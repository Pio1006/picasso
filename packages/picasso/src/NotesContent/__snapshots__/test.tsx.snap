// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesContent default render 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <p
      class="MuiTypography-root PicassoTypography-bodyMedium PicassoTypography-black MuiTypography-body1"
    >
      content
    </p>
  </div>
</div>
`;
